### PR TITLE
Recipe to remove Id annotation from Embeddable class when referenced by EmbeddedId

### PR DESCRIPTION
**Pull request by evie-lau**
## What's changed?
Add new recipe and tests for OpenJPA -> EclipseLink migration

Remove `@Id` annotation if...
- parent class is `@Embeddable` and
- parent class referenced by `@EmbeddedId`

## What's your motivation?
https://wiki.eclipse.org/EclipseLink/Examples/JPA/Migration/OpenJPA/Mappings#.40EmbeddedId_and_.40Id_in_the_Same_Entity

### Checklist
- [x] I've added unit tests to cover both positive and negative cases
- [x] I've read and applied the [recipe conventions and best practices](https://docs.openrewrite.org/authoring-recipes/recipe-conventions-and-best-practices)
- [x] I've used the IntelliJ IDEA auto-formatter on affected files

### CODE REVIEW

**Review by timtebeek — APPROVED** [review]
Implementation looks perfect to me; great to see! I've done some very minor touch ups to the docs, code highlighting inside text blocks, and formatting. Good to merge if you ask me, but let's wait for @cjobinabo to return before we merge.

File: src/test/java/org/openrewrite/java/migrate/javax/RemoveEmbeddableIdTest.java
Comment (cjobinabo) [inline src/test/java/org/openrewrite/java/migrate/javax/RemoveEmbeddableIdTest.java:27]: It would be good to include the following scenarios:

- `Class1` annotated with `@Entity` and defines a field of type `Class2` annotated with `@Embedded`. Then `Class2`'s definition is annotated with `@Embeddable` and defines a field of any type annotated with `@Id`
- `@Embeddable ` is used on a sub class where this subclass uses `@Id`

Comment (evie-lau) [reply src/test/java/org/openrewrite/java/migrate/javax/RemoveEmbeddableIdTest.java:27]: Does this recipe also have to handle the `@Embedded` case? The rule only mentioned `@EmbeddableId`

Comment (evie-lau) [reply src/test/java/org/openrewrite/java/migrate/javax/RemoveEmbeddableIdTest.java:27]: In the second case, is `Class1` defining `Class2` or `Class2SubClass` with `@EmbeddableId`?
And should the recipe be expected to run?

File: src/test/java/org/openrewrite/java/migrate/javax/RemoveEmbeddableIdTest.java
Comment (cjobinabo) [inline src/test/java/org/openrewrite/java/migrate/javax/RemoveEmbeddableIdTest.java:174]: I should have been clearer about describing the sub-class test case. I was looking for something like this:
```
              import javax.persistence.Embeddable;
              import javax.persistence.Id;
              public class SomeClass {
                        @Embeddable
                        public class EmbeddableObject {
                            @Id
                            private int field;
                        }
              }
```

Comment (evie-lau) [reply src/test/java/org/openrewrite/java/migrate/javax/RemoveEmbeddableIdTest.java:174]: If `SomeClass` also had it's own `@Id` annotated field, should that be left unchanged by this recipe?

Comment (evie-lau) [reply src/test/java/org/openrewrite/java/migrate/javax/RemoveEmbeddableIdTest.java:174]: Conversely, if the outer `SomeClass` is `@Embeddable`, has an `@Id` field, and has an inner class without `@Embeddable` and an inner field with `@Id`, should that inner field annotation be removed as well?

**Review by cjobinabo — APPROVED** [review]
Thanks for making the updates

**Pull request merged**
